### PR TITLE
VisualVM: new port

### PR DESCRIPTION
**Pull request by lhaeger**
###### Tested on

macOS 10.15.7 19H512
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: java/VisualVM/Portfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+name                VisualVM
```
Comment (Janosch) [inline java/VisualVM/Portfile:6]: This line could be omitted if you replace _visualvm_ with _VisualVM_ in the github.setup line

**Comment by lhaeger** [comment]
Maybe, but I live in a case-exact world and would like to keep that principle up, if you don't mind. :-)

File: java/VisualVM/Portfile (hunk 1 above)
Comment (lhaeger) [reply java/VisualVM/Portfile:6]: Maybe, but I live in a case-exact world and would like to keep that principle up, if you don't mind. :-)

**Comment by Janosch** [comment]
> Maybe, but I live in a case-exact world and would like to keep that principle up, if you don't mind. :-)

I don't mind :)

The only two other things which caught my attention are the fact that you did not use the java port group and by doing so did not provide a minimum JDK version that VisualVM requires. The other thing is, that this port is a binary port as nothing gets compiled. For me as a user this goes against my expectation of what a macports port usually is. However, I dont know if there is some sort of a convention in macports to make users aware of the fact that a port is precompiled.

I am a macports newbie, so both comments maybe completly irrelevant :)

**Comment by lhaeger** [comment]
> The only two other things which caught my attention are the fact that you did not use the java port group and by doing so did not provide a minimum JDK version that VisualVM requires. 

Good catch! Thanks, fixed that.

> The other thing is, that this port is a binary port as nothing gets compiled. For me as a user this goes against my expectation of what a macports port usually is. 

While MacPorts *can* compile from source in most cases this is not possible or feasible in others. MacPorts is also not only about building software from source. That's a key feature, yes, but it's also a software catalogue, helping people to identify suitable software. And it's also an software management tool that helps people to easily install/uninstall MacPorts as well as keep their software collection up to date.  Those use cases remain regardless of how the code was build and are the main reason why ports like these exist in the first place. 

> However, I don't know if there is some sort of a convention in macports to make users aware of the fact that a port is precompiled.

There are ongoing discussions about that, which have not yet produced applicable results. Until that happens, I'll stick to how it's been done with similar ports.

**Pull request merged**